lib/hal/video.c

```diff
@@ -2,6 +2,8 @@
 #include <hal/xbox.h>
 #include <hal/video.h>
 #include <string.h>
+#include <assert.h>
+#include <stdbool.h>
```

Comment:
This adds newer C features to this file, is this okay?
Should our driver code even use libc?

I'd argue this is fine; but we should have a minimalistic `assert` that uses `RtlAssert` in the future (to avoid side-effects of driver code).

Reply:
I think it's fine. The functionality it uses is rather minimal and easily stubbed should someone need to get rid of the libc dependency.

Using `RtlAssert` for `assert` is already possible by building PDCLib with `DEBUG_CONSOLE` defined. It's not the default to make it easier to use for users of a retail BIOS - we should probably have code decide on runtime which `assert` implementation is to be used.